CommunityToolkit.HighPerformance/Streams/MemoryStream{TSource}.cs

```diff
@@ -119,9 +119,13 @@ public sealed override Task CopyToAsync(Stream destination, int bufferSize, Canc
 
         try
         {
-            CopyTo(destination, bufferSize);
+            MemoryStream.ValidateDisposed(this.disposed);
 
-            return Task.CompletedTask;
+            Memory<byte> source = this.source.Memory.Slice(this.position);
+
+            this.position += source.Length;
+
+            return destination.WriteAsync(source, cancellationToken).AsTask();
```

Comment:
Should the order be swapped?
If any exception occurs, then position won't be updated, which is what's expected?

Reply:
I implemented this based on the existing `CopyTo` implementation, which itself seems to be inconsistent with the `Read` and `Write` overloads. Should I fix the original within this PR as well?

Reply:
The order there is following what `MemoryStream` does in the BCL ([see here](https://github.com/dotnet/runtime/blob/7adbc10dfb865fd5ecc37e309caa2d955b6061b0/src/libraries/System.Private.CoreLib/src/System/IO/MemoryStream.cs#L368)). The `Read` and `Write` APIs read/write first and then update the position, whereas `CopyTo` and `CopyToAsync` first update the position and then defer to the destination stream. We're just following suit with that same logic.